src/Sentry/Protocol/Envelopes/EnvelopeItem.cs

```diff
@@ -76,6 +76,21 @@ public EnvelopeItem(IReadOnlyDictionary<string, object?> header, ISerializable p
             var value => Convert.ToInt64(value) // can be int, long, or another numeric type
         };
 
+    internal long? TryGetOrRecalculateLength()
```

Comment:
I don't understand how we get here or how that fixes the issue - even if it does (which I believe). 
The attachment size did not get read correctly from the header? How/why not? Should it have been set in the first place? And if we're recalculating it here, should it not end up being set now?

Reply:
I'd put all of that in the details for the issue itself... but I just moved it into the details of this PR.

Reply:
So the length ends up in the header during serialization and gets removed during deserialization because of trust-issues. Fair enough.
But is this not a bug in the caching transport then? Should the length not be set back on the header during serialization?

Reply:
I'm not trying to bikeshed here, I genuinely don't get how it works.

Reply:
Yeah you have to read Matt's PR where he removes the header upon deserialisation in order to get it. It's linked to in the comment above the line where it gets removed.